x-pack/plugins/enterprise_search/server/routes/app_search/crawler.ts

```diff
@@ -27,6 +27,31 @@ export function registerCrawlerRoutes({
     })
   );
 
+  router.post(
+    {
+      path: '/api/app_search/engines/{name}/crawler/domains',
+      validate: {
+        params: schema.object({
+          name: schema.string(),
+        }),
+        body: schema.object({
+          name: schema.string(),
+          entry_points: schema.arrayOf(
+            schema.object({
+              value: schema.string(),
+            })
+          ),
+        }),
+        query: schema.object({
+          respond_with: schema.maybe(schema.string()),
+        }),
+      },
+    },
+    enterpriseSearchRequestHandler.createRequest({
+      path: '/api/as/v0/engines/:name/crawler/domains',
+    })
+  );
+
```

Comment:
Optional: I like to skip body validation when I can and make it pass through to the enterprise search server. Easier to write and easier to maintain.

```suggestion
  router.post(
    skipBodyValidation({
      path: '/api/app_search/engines/{name}/crawler/domains',
      validate: {
        params: schema.object({
          name: schema.string(),
        }),
        query: schema.object({
          respond_with: schema.maybe(schema.string()),
        }),
      },
    }),
    enterpriseSearchRequestHandler.createRequest({
      path: '/api/as/v0/engines/:name/crawler/domains',
    })
    
  );

```

Reply:
I've decided to leave this as is for now, but I'm interested in discussing this with the team.

Reply:
++, it's annoying to me that our internal endpoints aren't really documented and have to be inspected manually every time. I like using either the server or types to hold that information